main/src/ca/uwaterloo/flix/language/ast/Kind.scala

```diff
@@ -119,8 +120,32 @@ object Kind {
   }
 
   /**
-    * Returns a fresh kind variable.
+    * Returns the base of an arrow kind.
+    */
+  @tailrec
+  def base(k: Kind): Kind = k match {
+    case Arrow(k1, _) => base(k1)
+    case _ => k
+  }
+
+  /**
+    * Returns the arguments of an arrow kind.
     */
-  def freshVar()(implicit flix: Flix): Kind = Var(flix.genSym.freshId())
+  def args(k: Kind): List[Kind] = k match {
```

Comment:
In `Type` this is called `typeArgs`, maybe we should call it `kindArgs` for consistency.